### ISSUE TITLE
Cell View not using SwiftUI offset and custom gesture disables collection view gesture

### ISSUE THREAD

**Issue by alschmut** [amirdew/CollectionViewPagingLayout#55]
# Bug report
> First of all, thanks for creating this great library @amirdew!

I found some issue when using a dragGesture and an offset on the CollectionView cell. Below I laid out my observation and added a full example. 
- Have you noticed those issues already as well? In case it can not be fixed, it might be an idea to add that to the _limitations_ section.
- Is there another way of moving up/down an individual cell, to trigger some custom action, while the collection view is using the horizontal scroll direction?

Let me know if you have any related questions or need more information from my side!

### CellView does not rerender position when published offset changes
```swift
ScalePageView(...) { _ in
  CellView()
    .offset(...)
}
.options(.layout(.linear))
.gesture(...)
```
- does not update the cells view vertical position
- collectionView swipe gesture works
<hr>

### CollectionView swipe gesture is disabled
```swift
ScalePageView(...) { _ in
  CellView()
    .gesture(...)
    .offset(...)
}
.options(.layout(.linear))
```
- does not update the cells view vertical position
- disables/overrides the collectionView swipe gesture
<hr>

### Full example view
```swift
import SwiftUI
import CollectionViewPagingLayout

struct Item: Identifiable {
    let number: Int
    var id: Int {
        number
    }
}

let items = [
    Item(number: 0),
    Item(number: 1),
    Item(number: 2)
]

struct ContentView: View {
    @State var verticalOffset: CGFloat = 0
    
    var body: some View {
        ScalePageView(items) { _ in
            Rectangle()
                .fill(Color.orange)
                .offset(y: verticalOffset)
                .gesture(dragGesture)
        }
        .options(.layout(.linear))
        .pagePadding(.absolute(50))
    }
    
    var dragGesture: some Gesture {
        DragGesture()
            .onChanged {
                verticalOffset += $0.translation.height
            }
            .onEnded { _ in
                verticalOffset = 0
            }
    }
}
```

**Comment by amirdew** [comment]
Hey, thanks for reporting the issue.
Yes, it only refreshes the cells when `items` change. So, only if you change the order of items or the `id`.
But, I realized that's not convenient in your case. 
So I released a new version (1.0.3) now you can optionally make your model `Equatable`
Therefore, any change in your model will refresh the cells.

```swift
struct Item: Identifiable, Equatable {
    let number: Int
    var offset: CGFloat = 0
    var id: Int {
        number
    }
}

struct ContentView: View {
    @State var items = [
        Item(number: 0),
        Item(number: 1),
        Item(number: 2)
    ]
    
    @State var currentItemId: Item.ID? = 0
    
    var body: some View {
        ScalePageView(items, selection: $currentItemId) { item in
            Rectangle()
                .fill(Color.orange)
                .offset(y: item.offset)
        }
        .options(.layout(.linear))
        .pagePadding(.absolute(50))
        .gesture(dragGesture)
    }
    
    var dragGesture: some Gesture {
        DragGesture()
            .onChanged {
                items[currentItemId!].offset = $0.translation.height
            }
            .onEnded { _ in
                items[currentItemId!].offset = 0
            }
    }
}
```

**Comment by alschmut** [comment]
Thanks so much @amirdew! The two gestures now work amazingly well together ;)
